crates/bevy_utils/src/synccell.rs

```diff
@@ -0,0 +1,36 @@
+/// See [`Exclusive`](https://github.com/rust-lang/rust/issues/98407) for stdlib's upcoming implementation,
+/// which should replace this one entirely.
+///
+/// Provides a wrapper that allows making any type unconditionally [`Sync`] by only providing mutable access.
+#[repr(transparent)]
+pub struct SyncCell<T: ?Sized> {
+    inner: T,
+}
+
+impl<T: Sized> SyncCell<T> {
+    /// Construct a new instance of a `SyncCell` from the given value.
+    pub fn new(inner: T) -> Self {
+        Self { inner }
+    }
+
+    /// Deconstruct this `SyncCell` into its inner value.
+    pub fn to_inner(Self { inner }: Self) -> T {
+        inner
+    }
+}
+
+impl<T: ?Sized> SyncCell<T> {
+    /// Get a reference to this `SyncCell`'s inner value.
+    pub fn get(&mut self) -> &mut T {
+        &mut self.inner
+    }
+
+    /// Build a mutable reference to a `SyncCell` from a mutable reference
+    /// to its inner value, to skip constructing with [`new()`](SyncCell::new()).
+    pub fn from_mut(r: &'_ mut T) -> &'_ mut SyncCell<T> {
+        // SAFETY: repr is ≥ C, so refs have the same layout; and `SyncCell` properties are `&mut`-agnostic
```

Comment:
I don't understand the >= C bit? The `repr` is transparent here?

Reply:
Me neither, actually, I just pulled that from the `Exclusive` implementation and assumed it meant something to the lang devs.

Reply:
You can just change it to say "repr is transparent", it still applies.